libmodernize/main.py

```diff
@@ -28,6 +29,14 @@ def format_usage(usage):
     return usage
 
 def main(args=None):
+    old_os_linesep = os.linesep
+    try:
+        return actual_main(args)
+    finally:
+        if sys.version_info < (3, 0):
+            os.linesep = old_os_linesep
```

Comment:
On Python 3 we never change ``os.linesep``. Also, the monkeypatch to ``refactor._to_system_newlines`` should not cause any nondeterminism in the tests because it is always done prior to calling into ``lib2to3``, and is idempotent. (Also, it's only affecting lib2to3, unlike ``os.linesep`` which can affect other things.) Nevertheless, I can unpatch ``refactor._to_system_newlines`` if you think that would be an improvement.